mathics/core/atoms.py

```diff
@@ -36,18 +36,20 @@
 @lru_cache(maxsize=1024)
 def from_mpmath(value, prec=None):
     "Converts mpf or mpc to Number."
+    if isinstance(value, mpmath.mpc):
+        if value.imag == 0.0:
+            value = value.real
+        else:
+            real = from_mpmath(value.real, prec)
+            imag = from_mpmath(value.imag, prec)
+            return Complex(real, imag)
+
     if isinstance(value, mpmath.mpf):
```

Comment:
@TiagoCavalcante , also, you can improve a little bit more it, avoiding the `isinstance()` in case you already know that value is the imaginary part of an `mpmath.mpc`:

```
elif not isinstance(value, mpmath.mpf):
     raise TypeError(type(value))
if prec is None:
    return MachineReal(float(value))
else:
    return PrecisionReal(sympy.Float(str(value), prec))
```